.github/workflows/test.yml

```diff
@@ -561,5 +559,184 @@ jobs:
       - uses: codecov/codecov-action@v2
         with:
           files: ./packages/generator-helper/src/__tests__/coverage/clover.xml
-          flags: generator-helper,${{ matrix.os }}
+          flags: generator-helper,${{ matrix.os }},library,binary
+          name: generator-helper-${{ matrix.os }}
+
+  #
+  # Run all tests on macOS and Windows.
+  #
+  # Unlike the other jobs, this job doesn't use Docker (and thus skips some
+  # tests that require dependencies not easily installable without Docker —
+  # specifically, SQL Server).
+  #
+  # It also runs the tests for different packages sequentially to prevent the
+  # combinatorial explosion of the number of parallel jobs, to minimize the
+  # number of times we need to install MySQL using the package manager
+  # (PostgreSQL and MongoDB are provided by GitHub Actions out of the box), and
+  # minimize the time spent waiting for a free runner.
+  #
+  no-docker:
+    timeout-minutes: 60
+    runs-on: ${{ matrix.os }}
+    needs: detect_jobs_to_run
+
+    if: |
+      ${{
+        contains(needs.detect_jobs_to_run.outputs.jobs, '-all-') ||
+        contains(needs.detect_jobs_to_run.outputs.jobs, '-client-') ||
+        contains(needs.detect_jobs_to_run.outputs.jobs, '-migrate-') ||
+        contains(needs.detect_jobs_to_run.outputs.jobs, '-cli-')
+      }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-11, windows-latest]
+        node: [16]
+        queryEngine: ['library', 'binary']
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Prerequisites
+        shell: bash
+        run: |
+          echo "TEST_SKIP_MSSQL=true" >> $GITHUB_ENV
+          echo "TEST_MONGO_URI=mongodb://localhost:27017/tests" >> $GITHUB_ENV
+          echo "PRISMA_CLI_QUERY_ENGINE_TYPE=${{ matrix.queryEngine }}" >> $GITHUB_ENV
+          echo "PRISMA_CLIENT_ENGINE_TYPE=${{ matrix.queryEngine }}" >> $GITHUB_ENV
+
+      - name: Setup PostgreSQL
+        run: bash .github/workflows/setup-postgres.sh
+
+      - name: Setup MySQL
+        run: bash .github/workflows/setup-mysql.sh
+
+      - name: Cache .pnpm-store
+        uses: actions/cache@v1
+        with:
+          path: ~/.pnpm-store
+          key: ${{ runner.os }}-node${{ matrix.node }}-${{ hashFiles('**/pnpm-lock.yaml') }}
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+
+      - run: bash .github/workflows/setup.sh
+        env:
+          CI: true
+          SKIP_GIT: true
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+
+      - name: Test packages/sdk
+        if: ${{ contains(needs.detect_jobs_to_run.outputs.jobs, '-all-') }}
```

Comment:
```suggestion
        if: ${{ contains(needs.detect_jobs_to_run.outputs.jobs, '-all-') }} || contains(needs.detect_jobs_to_run.outputs.jobs, '-sdk-') }}
```

Reply:
FWIW, `detect-jobs-to-run.js` never produces `-sdk-`, it will output `-all-` in that case. Still a good idea to include the check here to future-proof it, yeah.